spec/checkers/foreign_key_checker_spec.rb

```diff
@@ -106,6 +140,210 @@
+        error_message: nil,
+        error_slug: nil
+      )
+    end
+  end
+
+  context 'when the association is covered by a composite foreign key in wrong order' do
+    before do
+      if adapter == 'mysql2' ||
+         ActiveRecord::VERSION::MAJOR < 7 ||
+         (ActiveRecord::VERSION::MAJOR == 7 && ActiveRecord::VERSION::MINOR < 1)
+        skip('Composite primary keys are supported only in Rails 7.1+')
+      end
+
+      define_database do
+        create_table :countries do |t|
+          t.string :code
+
+          t.index %i[code id], unique: true
+        end
+
+        create_table :entities do |t|
+          t.integer :country_id
+          t.string :country_code
+
+          t.foreign_key :countries, column: %i[country_code country_id], primary_key: %i[code id]
+        end
+      end
+    end
+
+    let(:country_class) do
+      define_class('Country', :countries) do |klass|
+        klass.primary_key = %i[code id]
+      end
+    end
+
+    let(:entity_class) do
+      define_class do |klass|
+        if ActiveRecord::VERSION::MAJOR == 7
+          klass.belongs_to :country, query_constraints: %i[country_id country_code]
+        else
+          klass.belongs_to :country, foreign_key: %i[country_id country_code]
+        end
+      end
+    end
+
+    specify do
+      expect(checker.report).to have_attributes(
+        checker_name: 'ForeignKeyChecker',
+        table_or_model_name: entity_class.name,
+        column_or_attribute_name: 'country',
+        status: :fail,
+        error_message: nil,
+        error_slug: :missing_foreign_key
+      )
+    end
+  end
+
+  context 'when foreign key is composite but association is not' do
+    before do
+      if ActiveRecord::VERSION::MAJOR < 7 || (ActiveRecord::VERSION::MAJOR == 7 && ActiveRecord::VERSION::MINOR < 1)
+        skip('Composite primary keys are supported only in Rails 7.1+')
+      end
+
+      define_database do
+        create_table :countries do |t|
+          t.string :code
+
+          t.index %i[code id], unique: true
+        end
+
+        create_table :entities do |t|
+          if adapter == 'mysql2'
+            t.bigint :country_id
+          else
+            t.integer :country_id
+          end
+          t.string :country_code
+
+          t.foreign_key :countries, column: %i[country_code country_id], primary_key: %i[code id]
+        end
+      end
+    end
+
+    let(:country_class) do
+      define_class('Country', :countries) do |klass|
+        klass.primary_key = %i[code id]
+      end
+    end
+
+    let(:entity_class) do
+      define_class do |klass|
+        klass.belongs_to :country
+      end
+    end
+
+    specify do
+      expect(checker.report).to have_attributes(
+        checker_name: 'ForeignKeyChecker',
+        table_or_model_name: entity_class.name,
+        column_or_attribute_name: 'country',
+        status: :fail,
+        error_message: nil,
+        error_slug: :missing_foreign_key
+      )
+    end
```

Comment:
I believe this test case should be `:ok` as per my [explanation](https://github.com/djezzzl/database_consistency/pull/258#issuecomment-3332357699) in the PR conversation - this use-case was exactly my intention for opening the PR